### PR TITLE
Add an aria-label to the editor blocks.

### LINKED ISSUE #562: Content blocks need to be labeled
When navigating through content blocks, assistive technologies need to report and inform users about _what_ kind of block they just landed on. Content blocks need to be labeled someway, an din a way that screen readers can announce properly.

Currently, there's nothing that can be announced in the markup. Blocks are just always the same:

screen shot 2017-04-25 at 00 48 35

Maybe, and aria-label attribute on the focusable blocks (with an appropriate text to report the type of blocks) could help. This should be carefully tested though.

This relates also to #557 where the block mover should announce _which_ kind of block it relates to.

Maybe, each block should carry a set of meta information available to be reused in other context. I see the state contains information about the blocks UID so maybe it should also contain a human-understandable name for each kind of blocks, to start with.

### PR DESCRIPTION
**Pull request by afercia**
First pass to try to add an aria-label to the Editor blocks. Code and wording could be improved, any feedback welcome.

All the editor blocks need to be labeled to allow screen reader users to get `what` a block is.

Screenshots:

screen shot 2017-06-30 at 17 55 40

screen shot 2017-06-30 at 17 57 09

Re: VoiceOver in the screenshot above:
- "group" is added by VoiceOver because blocks have no semantic role, so it announces them as generic "group"
- also the number of items in the group is added by VoiceOver

Fixes #562

### CODE REVIEW

File: editor/modes/visual-editor/block.js [hunk 1]
```diff
@@ -236,6 +237,7 @@ class VisualEditorBlock extends Component {
 	render() {
 		const { block, multiSelectedBlockUids } = this.props;
 		const blockType = getBlockType( block.name );
+		const blockLabel = sprintf( __( 'Editor block: %s' ), blockType.title );
```
Comment (youknowriad) [inline editor/modes/visual-editor/block.js]: Maybe I'd just use `Block: %s` but I'm fine either way

**Review by youknowriad — APPROVED** [review]
LGTM 👍

**Comment by afercia** [comment]
I've simplified the aria label to be just `Block: %s`. Also, added a translators comment, using the same one used for the block mover label. That could be probably improved a bit. Worth noting these translators comments should preferably be inside `sprintf()`. I see this has already been considered and there are thoughts to improve a bit the tooling, see https://github.com/WordPress/gutenberg/pull/984#discussion_r120464365
Not sure if there's already a specific GH issue.

**Comment by aduth** [comment]
>Edit: seems it has already been addressed? /cc @aduth

Yes, should be improved as of #1068

**Pull request merged**